### DISCUSSION TITLE
Load Ignition config to ISO

### DISCUSSION THREAD

**Discussion by fgamundi** [flatcar/Flatcar#1574] [Q&A]
Hi! I'm trying to modify the ISO image to preload some binaries and an Ignition configuration file.
[The documentation](https://www.flatcar.org/docs/latest/installing/bare-metal/booting-with-iso/) mentions "There is no straightforward way to provide an Ignition config." but doesn't mention how can this be achieved. I've tried multiple things, including modifying the syslinux configs at `/boot/syslinux/root.*.cfg` adding this kernel parameter `ignition.config.url=https://...` (also tried `file:// with a file in the ISO) and nothing works.

Any help appreciated. Thanks!

**Comment by tormath1** [comment]
Hi,

For `ignition.config.url` to work you need to set `flatcar.oem.id=metal` (the ISO image is using `flatcar.oem.id=qemu`) :) 

Is there a specific reason to use the ISO image?

**Reply by fgamundi** [reply]
We are using custom Flatcar images to be able to run our docker containers (also preloaded) on boot on multiple types of machines (cloud, vms, bare metal), and ISOs is one of the options we wanted to add to easily boot on a normal computer.

I've also added `flatcar.oem.id=metal` to `/boot/syslinux/root.*.cfg` and `boot/grub/menu.lst.*` files, trying an `https://`, `file://` and `oem://` config urls but it doesn't seem to work.

Sorry but I'm a bit lost here

Here's my `syslinux` file
```
label coreos.A
  menu label coreos.A
  kernel vmlinuz.A
  append console=ttyS0,115200n8 console=tty0 ro noswap cros_legacy  root=LABEL=ROOT rootflags=subvol=root usr=PARTLABEL=USR-A flatcar.oem.id=metal ignition.config.url=oem:///base/base.ign
```
And grub `menu.lst` file
```
timeout         0

title           Flatcar A Root
root            (hd0,0)
kernel          /syslinux/vmlinuz.A console=tty0 ro noswap cros_legacy  console=hvc0 root=LABEL=ROOT rootflags=subvol=root usr=PARTLABEL=USR-A flatcar.oem.id=metal ignition.config.url=oem:///base/base.ign

title           Flatcar B Root
root            (hd0,0)
kernel          /syslinux/vmlinuz.B console=tty0 ro noswap cros_legacy  console=hvc0 root=LABEL=ROOT rootflags=subvol=root usr=PARTLABEL=USR-B flatcar.oem.id=metal ignition.config.url=oem:///base/base.ign
```

**Reply by tormath1** [reply]
Once the instance booted, can you check the actual cmdline? `/proc/cmdline` and the actual OEM ID `/run/ignition.env` ?

**Reply by fgamundi** [reply]
Sure!
`/proc/cmdline`
```
rootflags=rw mount.usrflags=ro initrd=/flatcar/cpio.gz flatcar.autologin BOOT_IMAGE=/flatcar/vmlinuz
```
`/run/ignition.env`
```
OEM_ID=metal
PLATFORM_ID=metal
```

**Reply by tormath1** [reply]
Thanks, so the OEM_ID looks good - did you get an actual error or it just boots fine except that provisioning is not applied? Keep in mind that Ignition is only executed at first boot (you need to use a fresh image at each attempt)

**Reply by fgamundi** [reply]
I'm not getting (or at least I don't see) any errors. I'm testing the ISO in virtualbox without any disks attached, so it should be first boot everytime

**Reply by tormath1** [reply]
I'm still investigating - I think there is a deeper issue with this (that would explain the few ticket opened like https://github.com/flatcar/Flatcar/issues/51) - as a workaround, that would be possible to run `flatcar-install` once the ISO booted to install Flatcar on disk, when running this helper you can provide an Ignition file to execute when the instance will boot on disk. It's not ideal I know for your usecase, as said I'll continue to check this.
